### PR TITLE
OBSDOCS-1155: Update the warning admonition in the Monitoring docs wi…

### PR DESCRIPTION
**Pull request by eromanova97**
Version(s): `enterprise-4.12` and later

Issue: [OBSDOCS-1155](https://issues.redhat.com/browse/OBSDOCS-1155)

Link to docs preview:  https://78071--ocpdocs-pr.netlify.app/openshift-dedicated/latest/observability/monitoring/configuring-the-monitoring-stack.html

QE review:
- [x] QE has approved this change.

**Additional information:**

### CODE REVIEW

File: modules/monitoring-configuring-a-persistent-volume-claim.adoc [hunk 1]
```diff
@@ -141,11 +141,9 @@ data:
...
 ====
 
-. Save the file to apply the changes. The pods affected by the new configuration are restarted automatically and the new storage configuration is applied.
-ifdef::openshift-dedicated,openshift-rosa[]
+. Save the file to apply the changes. The pods affected by the new configuration are automatically redeployed and the new storage configuration is applied.
 +
 [WARNING]
 ====
-When changes are saved to a monitoring config map, the pods and other resources in the related project might be redeployed. The running monitoring processes in that project might also be restarted.
+When you update the config map with a PVC confiuration, the affected `StatefulSet` object is recreated, resulting in a temporary service outage."
```
Comment (eromanova97) [inline modules/monitoring-configuring-a-persistent-volume-claim.adoc:148]: @juzhao Please correct me if this is wrong, but when I tested this, the statefulset got redeployed in the same way it does when I resized the PV.  So I included the same warning here as in the resizing section.

File: modules/monitoring-creating-cluster-id-labels-for-metrics.adoc [hunk 2]
```diff
@@ -157,10 +157,4 @@ data:
...
 <3> The `replace` write relabel action replaces the temporary label with the target label for outgoing metrics. This action is the default and is applied if no action is specified.
 
-. Save the file to apply the changes to the `ConfigMap` object.
-The pods affected by the updated configuration automatically restart.
-+
-[WARNING]
-====
-Saving changes to a monitoring `ConfigMap` object might redeploy the pods and other resources in the related project. Saving changes might also restart the running monitoring processes in that project.
-====
+. Save the file to apply the changes. The new configuration is applied automatically.
```
Comment (eromanova97) [inline modules/monitoring-creating-cluster-id-labels-for-metrics.adoc:160]: @juzhao  When I tested this, this change did not trigger pods to redeploy so I am not including the sentence about the redeployment here.

File: modules/monitoring-setting-the-body-size-limit-for-metrics-scraping.adoc [hunk 3]
```diff
@@ -52,10 +52,4 @@ Valid numeric values use the Prometheus data size format: B (bytes), KB (kilobyt
...
 You can also set the value to `automatic` to calculate the limit automatically based on cluster capacity.
 
-. Save the file to apply the changes automatically.
-+
-[WARNING]
-====
-When you save changes to a `cluster-monitoring-config` config map, the pods and other resources in the `openshift-monitoring` project might be redeployed.
-The running monitoring processes in that project might also restart.
-====
+. Save the file to apply the changes. The new configuration is applied automatically.
```
Comment (eromanova97) [inline modules/monitoring-setting-the-body-size-limit-for-metrics-scraping.adoc:55]: @juzhao  When I tested this, this change did not trigger pods to redeploy so I am not including the sentence about the redeployment here.

File: modules/monitoring-attaching-additional-labels-to-your-time-series-and-alerts.adoc [hunk 4]
```diff
@@ -135,8 +135,3 @@ data:
 ----
 
 . Save the file to apply the changes. The new configuration is applied automatically.
```
Comment (eromanova97) [inline modules/monitoring-attaching-additional-labels-to-your-time-series-and-alerts.adoc:137]: @juzhao  When I tested this, this change did not trigger pods to redeploy so I am not including the sentence about the redeployment here.

File: modules/monitoring-choosing-a-metrics-collection-profile.adoc [hunk 5]
```diff
@@ -64,4 +58,4 @@ data:
       collectionProfile: *minimal*
 ----
 
-. Save the file to apply the changes. The pods affected by the new configuration restart automatically.
\ No newline at end of file
+. Save the file to apply the changes. The new configuration is applied automatically.
```
Comment (eromanova97) [inline modules/monitoring-choosing-a-metrics-collection-profile.adoc:61]: @juzhao  When I tested this, this change did not trigger pods to redeploy so I am not including the sentence about the redeployment here.

File: modules/monitoring-configuring-a-persistent-volume-claim.adoc (hunk 1 above)
Comment (juzhao) [reply modules/monitoring-configuring-a-persistent-volume-claim.adoc:148]: thanks, you are right

Comment (juzhao) [inline modules/monitoring-configuring-a-persistent-volume-claim.adoc:148]:
```suggestion
When you update the config map with a PVC confiuration, the affected `StatefulSet` object is recreated, resulting in a temporary service outage.
```

File: modules/monitoring-attaching-additional-labels-to-your-time-series-and-alerts.adoc (hunk 4 above)
Comment (juzhao) [reply modules/monitoring-attaching-additional-labels-to-your-time-series-and-alerts.adoc:137]: correct, no pods redeployed, the change is updated to prometheus configuration
```
$ oc -n openshift-monitoring exec -c prometheus prometheus-k8s-0 -- cat /etc/prometheus/config_out/prometheus.env.yaml | head
global:
  evaluation_interval: 30s
  scrape_interval: 30s
  external_labels:
    environment: prod
    prometheus: openshift-monitoring/k8s
    prometheus_replica: prometheus-k8s-0
    region: eu
```

File: modules/monitoring-choosing-a-metrics-collection-profile.adoc (hunk 5 above)
Comment (juzhao) [reply modules/monitoring-choosing-a-metrics-collection-profile.adoc:61]: right, no pods restarted

File: modules/monitoring-creating-cluster-id-labels-for-metrics.adoc (hunk 2 above)
Comment (juzhao) [reply modules/monitoring-creating-cluster-id-labels-for-metrics.adoc:160]: yes, mentioned pods won't restart in https://issues.redhat.com/browse/OBSDOCS-1155

File: modules/monitoring-setting-the-body-size-limit-for-metrics-scraping.adoc (hunk 3 above)
Comment (juzhao) [reply modules/monitoring-setting-the-body-size-limit-for-metrics-scraping.adoc:55]: yes, not restarted, configuration loaded to prometheus
```
$ oc -n openshift-monitoring exec -c prometheus prometheus-k8s-0 -- cat /etc/prometheus/config_out/prometheus.env.yaml | grep body_size_limit | uniq
  body_size_limit: 40MB
```

File: modules/monitoring-resizing-a-persistent-volume.adoc [hunk 6]
```diff
@@ -137,9 +137,9 @@ data:
 Storage requirements for the `thanosRuler` component depend on the number of rules that are evaluated and how many samples each rule generates.
 ====
 
-. Save the file to apply the changes.
+. Save the file to apply the changes. The pods affected by the new configuration are automatically redeployed.
 +
 [WARNING]
 ====
-When you save changes to a monitoring config map, the pods and other resources in the related project are redeployed. The monitoring processes running in that project are restarted.
+When you update the config map with a new storage size, the affected `StatefulSet` object is recreated, resulting in a temporary service outage."
```
Comment (juzhao) [inline modules/monitoring-resizing-a-persistent-volume.adoc:144]:
```suggestion
When you update the config map with a new storage size, the affected `StatefulSet` object is recreated, resulting in a temporary service outage.
```

File: modules/monitoring-investigating-why-user-defined-metrics-are-unavailable.adoc [hunk 7]
```diff
@@ -132,12 +132,7 @@ data:
 # ...
 ----
 +
-.. Save the file to apply the changes.
-+
-[NOTE]
-====
-The `prometheus-operator` in the `openshift-user-workload-monitoring` project restarts automatically when you apply the log-level change.
-====
+.. Save the file to apply the changes. The new configuration is applied automatically.
```
Comment (juzhao) [inline modules/monitoring-investigating-why-user-defined-metrics-are-unavailable.adoc:135]: no need to change
apply debug loglevel for `prometheus-operator`, `prometheus-operator` pod would recreate
```
$ oc -n openshift-user-workload-monitoring get pod
NAME                                   READY   STATUS    RESTARTS   AGE
alertmanager-user-workload-0           6/6     Running   0          178m
alertmanager-user-workload-1           6/6     Running   0          178m
prometheus-operator-6c6d7d75bc-knbgb   2/2     Running   0          31s
prometheus-user-workload-0             6/6     Running   0          3h11m
prometheus-user-workload-1             6/6     Running   0          3h11m
thanos-ruler-user-workload-0           4/4     Running   0          178m
thanos-ruler-user-workload-1           4/4     Running   0          178m

$ oc -n openshift-user-workload-monitoring get pod prometheus-operator-6c6d7d75bc-knbgb -oyaml | grep log-level
    - --log-level=debug
```
maybe we can change the original note to
The `prometheus-operator` pod in the `openshift-user-workload-monitoring` project restarts automatically when you apply the log-level change.

just add `pod`

Comment (eromanova97) [reply modules/monitoring-investigating-why-user-defined-metrics-are-unavailable.adoc]: Thank you, addressed. I changed it to:
```
The affected `prometheus-operator` pod is automatically redeployed.
```
to be consistent with the rest of the docs :+1:

File: modules/monitoring-configuring-the-monitoring-stack.adoc [hunk 8]
```diff
@@ -136,9 +136,21 @@ The Prometheus config map component is called `prometheusK8s` in the `cluster-mo
...
 
-. Save the file to apply the changes to the `ConfigMap` object. The pods affected by the new configuration are restarted automatically.
+. Save the file to apply the changes to the `ConfigMap` object.
 +
 [WARNING]
 ====
-When changes are saved to a monitoring config map, the pods and other resources in the related project might be redeployed. The running monitoring processes in that project might also be restarted.
+Different configuration changes to the `ConfigMap` object result in different outcomes:
+
+* The pods are not redeployed, therefore, there is no service outage.
```
Comment (lpettyjo) [inline modules/monitoring-configuring-the-monitoring-stack.adoc:145]: Change to: "The pods are not redeployed. Therefore, there is no service outage."

**Review by lpettyjo — COMMENTED** [review]
Otherwise, LGTM!

**Review by sheriff-rh — APPROVED** [review]
Changes look good; merging to 4.12+.

**Comment by sheriff-rh** [comment]
@eromanova97 you will need to manually cherry pick changes to 4.12 and 4.13. Be sure to include this PR in the top comment for tracking.

**Comment by eromanova97** [comment]
Thank you @sheriff-rh the manual cherry-picks are ready:
https://github.com/openshift/openshift-docs/pull/78636
https://github.com/openshift/openshift-docs/pull/78637

**Comment by sheriff-rh** [comment]
Done, thank you @eromanova97 !

**Pull request merged**